index.bs

```diff
@@ -332,37 +360,49 @@ Reporting paint timing {#sec-reporting-paint-timing}
 
-    1. If |document| [=should report first contentful paint=], then:
-        1. Invoke the [[#report-paint-timing]] algorithm with |document|, <code>"first-contentful-paint"</code>, and |paintTimestamp| as arguments.
+            NOTE: A [=document=] is not guaranteed to mark <code>"first-paint"</code> or <code>"first-contentful-paint"</code>. A completely blank [=document=] may never mark [=first paint=], and a [=document=] containing only elements that are not [=contentful=] may never mark [=first contentful paint=].
 
-        NOTE: A parent frame should not be aware of the paint events from its child iframes, and vice versa. This means that a frame that contains just iframes will have [=first paint=] (due to the enclosing boxes of the iframes) but no [=first contentful paint=].
+            NOTE: The marking of [=first paint=] is optional. User-agents implementing paint timing should at the very least mark [=first contentful paint=].
 
-        NOTE: A [=document=] is not guaranteed to mark <code>"first-paint"</code> or <code>"first-contentful-paint"</code>. A completely blank [=document=] may never mark [=first paint=], and a [=document=] containing only elements that are not [=contentful=] may never mark [=first contentful paint=].
+        1. [=Report largest contentful paint=] given |document|, |paintTimingInfo|,
+            |paintedImages| and |paintedTextNodes|.
+        1. [=Report element timing=] given |document|, |paintTimingInfo|,
+            |paintedImages| and |paintedTextNodes|.
 
-        NOTE: The marking of [=first paint=] is optional. User-agents implementing paint timing should at the very least mark [=first contentful paint=].
+    1. If the user-agent does not support implementation-defined presentation times, call |flushPaintTimings| and return.
 
-    1. [=Report largest contentful paint=] given |document|, |paintTimestamp|,
-        |paintedImages| and |paintedTextNodes|.
-    1. [=Report element timing=] given |document|, |paintTimestamp|,
-        |paintedImages| and |paintedTextNodes|.
+    1. Run the following steps [=In parallel=]:
+        1. Wait until an implementation-defined time when the current frame has been presented to the user.
+        1. Set |paintTimingInfo|'s [=implementation-defined presentation time=] to the [=current high resolution time=] given |document|'s [=relevant global object=].
+        1. If |document|'s [=environment settings object/cross-origin isolated capability=] is false, then:
+            1. Coarsen |paintTimingInfo|'s [=implementation-defined presentation time=] to the next multiple of 4 milliseconds, or coarser.
+            1. Wait until the [=current high resolution time=] is |paintTimingInfo|'s [=implementation-defined presentation time=].
+        1. [=Queue a global task=] on the [=performance timeline task source=] given |document|'s [=relevant global object=] to |flushPaintTimings|.
```

Comment:
This should probably read "`to execute |flushPaintTimings|`"

(or "call", "run", etc... but I think you need a verb.)

Reply:
Done